chapter05/src/main/java/com/seaofnodes/simple/Parser.java

```diff
@@ -479,7 +479,7 @@ private boolean isPunctuation(char ch) {
 
         private String parsePunctuation() {
             int start = _position;
-            return new String(_input, start, start+1);
+            return new String(_input, start, 1);
```

Comment:
punctuations can be more than 1 char long - e.g. ==

Reply:
This is used for error messages where it is unclear how long the punctuation shoould be. I think using only on character here is fine.

Reply:
well, the original code mis-used the new String API, and had the position being used as the length, so I think this is a clear winner.